contracts/Witch.sol

```diff
@@ -128,6 +131,7 @@ contract Witch is AccessControl() {
         external
         returns (uint256 ink)
     {
+        require (auctions[vaultId].start > 0, "Vault not under auction");
```

Comment:
The only bit that could be in a shared function are the first few lines where the data is pulled from storage. There are subtle differences in the rest that don't allow to share any more implementation meaningfully. If it would be copied in any more places, or if it would have any more complexity to it, I'd do it. As it stands, I don't think it's worth it.